modules/caddyhttp/fileserver/staticfiles.go

```diff
@@ -639,6 +659,23 @@ func calculateEtag(d os.FileInfo) string {
 	return `"` + t + s + `"`
 }
 
+// Finds the first corresponding etag file for a given file in the file system and return its content
+func (fsrv *FileServer) getEtagFromFile(fileSystem fs.FS, filename string) (string, error) {
+	for _, suffix := range fsrv.EtagFileExtensions {
+		etagFilename := filename + suffix
+		_, err := fs.Stat(fileSystem, etagFilename)
+		if err != nil {
+			continue
+		}
```

Comment:
Is the purpose of this to check whether we can access the file? Maybe we can eliminate this system call and instead just call `fs.ReadFile()`. Then:

```go
if errors.Is(err, fs.ErrNotExist) || errors.Is(err, fs.ErrPermission) {
    continue
}
if err != nil {
    return "", fmt.Errorf("cannot read etag from file %s: %v", etagFilename, err)
}
```

What do you think?

Reply:
Much much better. Just changed it. Thanks!